data-structures/queue/queue.py

```diff
@@ -0,0 +1,46 @@
+from .node import Node
+
+
+class Queue:
+    def __init__(self, iterable=[]):
+        """ testing for an empty queue"""
+        self.front = None
+        self.back = None
+        self._len = 0
+
+        for item in iterable:
+            self.enqueue(item)
+
+    def __len__(self):
+        """ return the length of the queue """
+        return self._len
+
+    def __str__(self):
+        """ return the items in the queue """
+        string = ""
+        current = self.front
+        for _ in range(self._len):
+            string += str(current.val) + " "
+            current = current._next
+        return string.rstrip()
+
+    def enqueue(self, val):
+        """ adds to a queue much like push """
+        node = Node(val)
+        if self._len == 0:
+            self.back = node
+            self.front = node
+        else:
+            self.back = node
```

Comment:
Currently this is broken because you are not connecting the old `back` node to the new `back` node. Thus the `front` and `back` are no longer connected to each other.